src/server/system_services/bucket_server.js

```diff
@@ -1809,6 +1846,59 @@ async function put_object_lock_configuration(req) {
     });
 }
 
+function _is_bucket_nsfs_exported(bucket) {
+    return bucket.namespace && bucket.namespace.write_resource &&
+        bucket.namespace.write_resource.nsfs_config;
+}
+
+// 1. create namespace resource 
+// 2. create namespace bucket
+async function export_dir_as_bucket(req) {
```

Comment:
I would avoid adding this new api which just wraps the two calls specific to nsfs - the creation of a namespace resource + namespace bucket is a general problem that we created by having namespace resource represent a specific single bucket/dir. I think that what we should focus on doing is to allow namespace resources to be non-specific and only define the base fs_path (or base s3/blob account) and then namespace buckets will be able to reuse those resources just like we do for pools.

Adding more automation to create/delete the resource automatically with the bucket is just digging the hole deeper for these redundant configuration objects.

Reply:
removed.